File: summary/ruby/4_bubble_sort.rb
def bubble_sort(arr)
  n = arr.length

  (0...n).each do |i|
    (0...n - i - 1).each do |j|
      arr[j], arr[j + 1] = arr[j + 1], arr[j] if arr[j] > arr[j + 1]
    end
  end

  arr
end

arr = gets.chomp.split.map(&:to_i)
sorted_arr = bubble_sort(arr)

puts "#{sorted_arr}"

File: summary/python/4_bubble_sort.py
def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
    return arr

# Example usage:
input_array = list(map(int, input().split()))
sorted_array = bubble_sort(input_array)
print("Sorted Array:", sorted_array)
